javascript/sbp/observation.js

```diff
@@ -0,0 +1,1016 @@
+  buffers.push(b);
+  b = new Buffer( 8 );
+  b.writeDoubleLE(this.fields.toe_tow);
+  buffers.push(b);
+  b = new Buffer( 2 );
+  b.writeUInt16LE(this.fields.toe_wn);
+  buffers.push(b);
+  b = new Buffer( 8 );
+  b.writeDoubleLE(this.fields.toc_tow);
+  buffers.push(b);
+  b = new Buffer( 2 );
+  b.writeUInt16LE(this.fields.toc_wn);
+  buffers.push(b);
+  b = new Buffer( 1 );
+  b.writeUInt8(this.fields.valid);
+  buffers.push(b);
+  b = new Buffer( 1 );
+  b.writeUInt8(this.fields.healthy);
+  buffers.push(b);
+  b = new Buffer( 1 );
+  b.writeUInt8(this.fields.prn);
+  buffers.push(b);
+  b = new Buffer( 1 );
+  b.writeUInt8(this.fields.iode);
+  buffers.push(b);
+  return Buffer.concat(buffers);
+};
+MSG_EPHEMERIS_DEP_B.prototype.parser = new Parser()
+  .endianess('little')
+  .doublele('tgd')
+  .doublele('c_rs')
+  .doublele('c_rc')
+  .doublele('c_uc')
+  .doublele('c_us')
+  .doublele('c_ic')
+  .doublele('c_is')
+  .doublele('dn')
+  .doublele('m0')
+  .doublele('ecc')
+  .doublele('sqrta')
+  .doublele('omega0')
+  .doublele('omegadot')
+  .doublele('w')
+  .doublele('inc')
+  .doublele('inc_dot')
+  .doublele('af0')
+  .doublele('af1')
+  .doublele('af2')
+  .doublele('toe_tow')
+  .uint16('toe_wn')
+  .doublele('toc_tow')
+  .uint16('toc_wn')
+  .uint8('valid')
+  .uint8('healthy')
+  .uint8('prn')
+  .uint8('iode');
+
+var PackedObsContentDepA = function (sbp, msg) {
+  
+  /*SBP class for message fragment PackedObsContentDepA
+
+  
+  Deprecated.
+
+  Parameters
+  ----------
+  P : int
+    Pseudorange observation
+  L : CarrierPhase
+    Carrier phase observation
+  cn0 : int
+    Carrier-to-Noise density
+  lock : int
+    Lock indicator. This value changes whenever a satellite
+signal has lost and regained lock, indicating that the
+carrier phase ambiguity may have changed.
+
+  prn : int
+    PRN-1 identifier of the satellite signal
+  sender : int
+    Optional sender ID, defaults to SENDER_ID (see sbp/msg.py).
+
+  */
+
+  this.sbp = sbp;
+  this.fields = this.parser.parse(msg);
+
+  return this;
+};
+PackedObsContentDepA.prototype = Object.create(SBP.prototype);
+PackedObsContentDepA.prototype.constructor = PackedObsContentDepA;
+PackedObsContentDepA.prototype.toBuffer = function () {
+  var buffers = [];
+  var b;
+  b = new Buffer( 4 );
+  b.writeUInt32LE(this.fields.P);
+  buffers.push(b);
+  buffers = buffers.concat( CarrierPhase.prototype.toBuffer.apply({ fields: this.fields.L }));
+  b = new Buffer( 1 );
+  b.writeUInt8(this.fields.cn0);
+  buffers.push(b);
+  b = new Buffer( 2 );
+  b.writeUInt16LE(this.fields.lock);
+  buffers.push(b);
+  b = new Buffer( 1 );
+  b.writeUInt8(this.fields.prn);
+  buffers.push(b);
+  return Buffer.concat(buffers);
+};
+PackedObsContentDepA.prototype.parser = new Parser()
+  .endianess('little')
+  .uint32('P')
+  .nest('L', { type: CarrierPhase.prototype.parser })
+  .uint8('cn0')
+  .uint16('lock')
+  .uint8('prn');
+
+var MSG_OBS_DEP_A = function (sbp, msg) {
```

Comment:
These object names should probably be cased (vs capped). All caps has typically been used for constants.